spec/system/bus_stops/managing_bus_stops_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe 'managing stops as an admin' do
+  let(:admin) { create :user, :admin }
+  let!(:bus_stop) { create :bus_stop }
+  before :each do
+    when_current_user_is admin
+    visit manage_bus_stops_url
+  end
+  context 'delete button' do
+    it 'deletes the specific bus stop' do
+      expect(page).to have_selector 'table.manage tbody tr', count: 1
+      within 'tr', text: bus_stop.name do
```

Comment:
This is a cool way to do this, I haven't seen a `within` block in a test before. You mentioned that `bus_stop` names don't need to be unique though. Actually you set them to be the same at creation. So, this wouldn't actually help specify what button to press.

Reply:
You're right, will fix.